src/service/action-impl.js

```diff
@@ -176,7 +176,17 @@ export class ActionService {
       // fast-click.
       this.root_.addEventListener('click', event => {
         if (!event.defaultPrevented) {
-          this.trigger(dev().assertElement(event.target), 'tap', event);
+          this.trigger(dev().assertElement(event.target), name, event);
+        }
+      });
+      this.root_.addEventListener('keydown', event => {
+        if (event.keyCode == 13 /* enter */ ||
+            event.keyCode == 32 /* space */) {
+          const element = dev().assertElement(event.target);
+          if (!event.defaultPrevented &&
+              element.getAttribute('role') == 'button') {
+            this.trigger(element, name, event);
```

Comment:
ditto: comment from the other PR: `preventDefault()` or SPACE will scroll the page.